# This patch fixes the following error on smartos:

#In file included from /usr/include/sys/procfs.h:50,
#                 from /usr/include/sys/ucred.h:33,
#                 from /export/home/alasdair/s10-userland/components/postgresql91/postgresql-9.1.1/src/port/getpeereid.c:27:
# /usr/include/sys/old_procfs.h:52:2: error: #error "Cannot use procfs in the large file compilation environment"

diff -ruN postgresql-9.3.3.orig/configure.in postgresql-9.3.3/configure.in
--- postgresql-9.3.3.orig/configure.in	2014-02-17 19:29:55.000000000 +0000
+++ postgresql-9.3.3/configure.in	2014-02-26 10:37:19.246962424 +0000
@@ -19,10 +19,6 @@
 
 AC_INIT([PostgreSQL], [9.3.3], [pgsql-bugs@postgresql.org])
 
-m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.63], [], [m4_fatal([Autoconf version 2.63 is required.
-Untested combinations of 'autoconf' and PostgreSQL versions are not
-recommended.  You can remove the check from 'configure.in' but it is then
-your responsibility whether the result works or not.])])
 AC_COPYRIGHT([Copyright (c) 1996-2013, PostgreSQL Global Development Group])
 AC_CONFIG_SRCDIR([src/backend/access/common/heaptuple.c])
 AC_CONFIG_AUX_DIR(config)
@@ -993,13 +989,6 @@
 #endif
 ])
 
-# On OpenBSD, test for sys/ucred.h will fail unless sys/param.h
-# is included first.
-AC_CHECK_HEADERS(sys/ucred.h, [], [],
-[AC_INCLUDES_DEFAULT
-#include <sys/param.h>
-])
-
 # At least on IRIX, test for netinet/tcp.h will fail unless
 # netinet/in.h is included first.
 AC_CHECK_HEADERS(netinet/in.h)
